pimd/pim_mroute.h

```diff
@@ -30,153 +30,109 @@
 #define __EXTENSIONS__
 #endif
 
-#include <netinet/in.h>
-#ifdef HAVE_NETINET_IP_MROUTE_H
-#include <netinet/ip_mroute.h>
-#endif
 
 #define PIM_MROUTE_MIN_TTL (1)
 
 #if PIM_IPV == 4
+
+#include <netinet/in.h>
 #if defined(HAVE_LINUX_MROUTE_H)
 #include <linux/mroute.h>
 #else
-/*
-  Below: from <linux/mroute.h>
-*/
-
-#ifndef MAXVIFS
-#define MAXVIFS (256)
+#ifndef VTYSH_EXTRACT_PL
+#include "linux/mroute.h"
 #endif
-
-#ifndef SIOCGETVIFCNT
-#define SIOCGETVIFCNT   SIOCPROTOPRIVATE        /* IP protocol privates */
-#define SIOCGETSGCNT    (SIOCPROTOPRIVATE+1)
-#define SIOCGETRPF      (SIOCPROTOPRIVATE+2)
 #endif
 
-#ifndef MRT_INIT
-#define MRT_BASE     200
-#define MRT_INIT     (MRT_BASE)      /* Activate the kernel mroute code      */
-#define MRT_DONE     (MRT_BASE+1)    /* Shutdown the kernel mroute           */
-#define MRT_ADD_VIF  (MRT_BASE+2)    /* Add a virtual interface              */
-#define MRT_DEL_VIF  (MRT_BASE+3)    /* Delete a virtual interface           */
-#define MRT_ADD_MFC  (MRT_BASE+4)    /* Add a multicast forwarding entry     */
-#define MRT_DEL_MFC  (MRT_BASE+5)    /* Delete a multicast forwarding entry  */
-#define MRT_VERSION  (MRT_BASE+6)    /* Get the kernel multicast version     */
-#define MRT_ASSERT   (MRT_BASE+7)    /* Activate PIM assert mode             */
-#define MRT_PIM      (MRT_BASE+8)    /* enable PIM code      */
-#endif
+typedef struct mfcctl pim_mfcctl;
```

Comment:
these defs seem to be used only in very few places - move there where possible

Reply:
Addressed